pkg/ottl/parser_collection.go

```diff
@@ -37,92 +18,36 @@ type StatementsGetter interface {
-	return defaultStatementsGetter(statements)
-}
-
-// ottlParserWrapper wraps an ottl.Parser using reflection, so it can invoke exported
-// methods without knowing its generic type (transform context).
-type ottlParserWrapper struct {
-	parser                         reflect.Value
-	prependContextToStatementPaths func(context string, statement string) (string, error)
-}
-
-func newParserWrapper[K any](parser *Parser[K]) *ottlParserWrapper {
-	return &ottlParserWrapper{
-		parser:                         reflect.ValueOf(parser),
-		prependContextToStatementPaths: parser.prependContextToStatementPaths,
-	}
-}
-
-func (g *ottlParserWrapper) parseStatements(statements []string) (reflect.Value, error) {
-	method := g.parser.MethodByName("ParseStatements")
-	parseStatementsRes := method.Call([]reflect.Value{reflect.ValueOf(statements)})
-	err := parseStatementsRes[1]
-	if !err.IsNil() {
-		return reflect.Value{}, err.Interface().(error)
-	}
-	return parseStatementsRes[0], nil
+	return defaultStatementsConditionsGetter(statements)
 }
 
-func (g *ottlParserWrapper) prependContextToStatementsPaths(context string, statements []string) ([]string, error) {
-	result := make([]string, 0, len(statements))
-	for _, s := range statements {
-		prependedStatement, err := g.prependContextToStatementPaths(context, s)
-		if err != nil {
-			return nil, err
-		}
-		result = append(result, prependedStatement)
-	}
-	return result, nil
+// ConditionsGetter represents a set of conditions to be parsed.
+//
+// Experimental: *NOTE* this API is subject to change or removal in the future.
+type ConditionsGetter interface {
+	// GetConditions retrieves the OTTL conditions to be parsed
+	GetConditions() []string
 }
 
-// statementsConverterWrapper is a reflection-based wrapper to the ParsedStatementConverter function,
-// which does not require knowing all generic parameters to be called.
-type statementsConverterWrapper reflect.Value
-
-func newStatementsConverterWrapper[K any, R any](converter ParsedStatementConverter[K, R]) statementsConverterWrapper {
-	return statementsConverterWrapper(reflect.ValueOf(converter))
+// NewConditionsGetter creates a new ConditionsGetter.
+//
+// Experimental: *NOTE* this API is subject to change or removal in the future.
+func NewConditionsGetter(statements []string) ConditionsGetter {
+	return defaultStatementsConditionsGetter(statements)
 }
 
-func (s statementsConverterWrapper) call(
-	parserCollection reflect.Value,
-	ottlParser *ottlParserWrapper,
-	context string,
-	statements StatementsGetter,
-	parsedStatements reflect.Value,
-) (reflect.Value, error) {
-	result := reflect.Value(s).Call([]reflect.Value{
-		parserCollection,
-		ottlParser.parser,
-		reflect.ValueOf(context),
-		reflect.ValueOf(statements),
-		parsedStatements,
-	})
-
-	resultValue := result[0]
-	resultError := result[1]
-	if !resultError.IsNil() {
-		return reflect.Value{}, resultError.Interface().(error)
-	}
+type defaultStatementsConditionsGetter []string
```

Comment:
This name wont scale if we add other concepts (like `ParseValueExpression`).  Can we use a more generic name like `defaultOTTLGetter` or similar?